vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -125,74 +181,92 @@ export class WebRequestWorker
+    {
+        return this.makeWebRequest(url, ctx, true, retriesCount);
+    }
+
+    private reportTimeAnalytics(response: any, options: any, url: string, ctx: IAcquisitionWorkerContext, manualFinalTime: bigint | null = null): void
+    {
+        // Streamed responses return out bits of data to be piped, so this would record the end time as if only the first few bytes finished.
+        // Instead we can manually report this when the stream is finished.
+
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!manualFinalTime && options?.responseType === 'stream')
+        {
+            return;
+        }
+
         // Standard timeout time in NS : 60,000,000,000 is < than std max_safe_int_size: 9,007,199,254,740,991
         const timerPrecision = 2; // decimal places for timer result
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
         const startTimeNs = (response as any)?.startTime;
+
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        const finalTimeNs = (response as any)?.finalTime;
+        const finalTimeNs = manualFinalTime ?? (response as any)?.finalTime;
+
         let durationMs = '-1';
         if (startTimeNs && finalTimeNs && finalTimeNs - startTimeNs < Number.MAX_SAFE_INTEGER)
         {
             durationMs = (Number(finalTimeNs - startTimeNs) / 1000000).toFixed(timerPrecision);
         }
         else
         {
-            this.context.eventStream.post(new WebRequestTimeUnknown(`Timer for request failed. Start time: ${startTimeNs}, end time: ${finalTimeNs}`, durationMs, 'true', url, String(response.status)));
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            ctx.eventStream.post(new WebRequestTimeUnknown(`Timer for request failed. Start time: ${startTimeNs}, end time: ${finalTimeNs}`, durationMs, 'true', url, String(response?.status)));
         }
-        if (!response.cached)
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!(response?.cached))
         {
-            this.context.eventStream.post(new WebRequestTime(`Timer for request:`, durationMs, 'true', url, String(response.status)));
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            ctx.eventStream.post(new WebRequestTime(`Timer for request:`, durationMs, 'true', url, String(response?.status)));
         }
         else
         {
-            this.context.eventStream.post(new WebRequestCachedTime(`Cached Timer for request:`, durationMs, 'true', url, String(response.status)));
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            ctx.eventStream.post(new WebRequestCachedTime(`Cached Timer for request:`, durationMs, 'true', url, String(response?.status)));
```

Comment:
Is it possible to give real types for these cases?

Reply:
The String() constructor here will produce a real string type, so that is good, but good question :)

If you mean the response object, I dont trust any of the types that the web libraries may return as they have never followed the correct logic and bypassed the type system, returning types their API said they would never return. So sadly I dont trust it